streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -1352,19 +1358,4 @@ public Position getPosition() {
         }
     }
 
-    private static class TestDriverProducer extends StreamsProducer {
```

Comment:
This class isn't doing anything, right? I'm unclear on its purpose and figured I might as well clean it up from the TTD -- but again, happy to undo this change since it's not needed for the main fix